privatek8s.tf

```diff
@@ -169,6 +169,32 @@ resource "azurerm_kubernetes_cluster_node_pool" "infraciarm64" {
   tags = local.default_tags
 }
 
+
+resource "azurerm_kubernetes_cluster_node_pool" "infracictrlarm64" {
+  name                  = "arm64ctrl"
+  vm_size               = "Standard_D4pds_v5" # 4 vCPU, 16 GB RAM, local disk: 150 GB and 19000 IOPS
+  os_disk_type          = "Ephemeral"
+  os_disk_size_gb       = 150 # Ref. Cache storage size at https://learn.microsoft.com/en-us/azure/virtual-machines/dpsv5-dpdsv5-series#dpdsv5-series (depends on the instance size)
+  orchestrator_version  = local.kubernetes_versions["privatek8s"]
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.privatek8s.id
+  enable_auto_scaling   = true
+  min_count             = 1
+  max_count             = 3
+  zones                 = [1] # zone 1 is dedicated to ARM64 by azure
+  vnet_subnet_id        = data.azurerm_subnet.privatek8s_tier.id
+
+  node_taints = [
+    "jenkins=infra.ci.jenkins.io:NoSchedule",
+    "kubernetes.azure.com/scalesetpriority=spot:NoSchedule",
+    "kind=controller"
+  ]
```

Comment:
We need to rethink the taints/toleration:

- https://github.com/jenkins-infra/kubernetes-management/blob/24eb85de3ba3720d5638011ed8861d38a3a98dc7/config/jenkins_release.ci.jenkins.io.yaml#L228-L236 has a toleration for a `jenkins=release.ci.jenkins.io:NoSchedule` taint. The current `linuxpool` does not have any taint that is why these tolerations are working today
- I see `kind=controller`. But what about the 3 other "Jenkins services" currently running on the `linuxnodepool` (github-ops-comment, rss2twitter and the ircbot) ? Do you plan to schedule them on a distinct nodepool in the loing term (once we will be running full arm64)? Same question for the "side" services such as cert-manager, datadog cluster-agent, nginx ingresses ?

Reply:
you mean that we should have a taint not specific for controller but for "no agent pool" ? to be able to schedule all the others services on it ?

Reply:
> you mean that we should have a taint not specific for controller but for "no agent pool" ? to be able to schedule all the others services on it ?

I mean you have to define the expected goal before implementing with taints/tolerations (e.g. what target architecture are you envisioning?):

- Distinct node pools for controllers and for other "non agent" services (bots, technical services, etc.)?
- Or an unified node pool for all?

=> The sizing (instance size and min/max node) and taints/toleration will be different in both cases.
I believe checking the current sizing, resource usage and costs should be the arguments to decide